packages/openapi-generator/src/parser/operation-naming.ts

```diff
@@ -0,0 +1,178 @@
+  const [operationsWithExtensions, operationsWithoutExtensions] = partition(
+    operations,
+    ({ operation }) => !!operation[operationNameExtension]
+  );
+
+  validateUniqueExtensions(operationsWithExtensions);
+
+  const renamedOperationsWithExtensions = renameOperations(
+    operationsWithExtensions,
+    renameOperationWithExtension
+  );
+
+  const [namedOperations, unnamedOperations] = partition(
+    operationsWithoutExtensions,
+    ({ operation }) => !!operation.operationId
+  );
+
+  const renamedUnnamedOperations = renameOperations(
+    unnamedOperations,
+    ({ pathPattern, method }) =>
+      getOperationNameFromPatternAndMethod(pathPattern, method)
+  );
+
+  return [
+    ...renamedOperationsWithExtensions,
+    ...namedOperations,
+    ...renamedUnnamedOperations
+  ];
+}
+
+function validateUniqueExtensions(operations: OperationInfo[]): void {
+  operations.reduce((uniqueOperationNames: string[], operationInfo) => {
+    const operationName = renameOperationWithExtension(operationInfo);
+    if (uniqueOperationNames.includes(operationName)) {
+      throw new Error(
+        `Operation name '${operationInfo.operation[operationNameExtension]}' provided for '${operationNameExtension}' resolves to '${operationName}' and is not unique.`
+      );
+    }
+    return [...uniqueOperationNames, operationName];
+  }, []);
+}
+
+function renameOperationWithExtension({ operation }: OperationInfo): string {
+  const originalOperationName = operation[operationNameExtension];
+  if (!originalOperationName) {
+    // This should never happen
+    throw new Error(
+      'Could not rename operation based on extension. Extension is not defined.'
+    );
+  }
+  return camelCase(originalOperationName);
+}
+
+function renameOperations(
+  operations: OperationInfo[],
+  renameFn: (operationInfo: OperationInfo) => string
+): OperationInfo[] {
+  return operations.map(operationInfo => {
+    operationInfo.operation.operationId = renameFn(operationInfo);
+    return operationInfo;
+  });
+}
+
+/**
+ * Partition operations into a list of unique names that do not need to be changed and a list of operations where the name is duplicate or subject to change and therefore potentially duplicate.
+ * @param namedOperations All operations that have an operationId
+ * @returns an object containing the unique operation names and operations with (potentially) duplicate names
+ */
+function partitionNamedOperationsToUniqueAndDuplicate(
+  namedOperations: OperationInfo[]
+): {
+  uniqueOperationNames: string[];
+  operationsWithUniqueNames: OperationInfo[];
+  operationsWithDuplicateNames: OperationInfo[];
+} {
+  return namedOperations.reduce(
+    (
+      partitionedOperations: {
+        uniqueOperationNames: string[];
+        operationsWithUniqueNames: OperationInfo[];
+        operationsWithDuplicateNames: OperationInfo[];
+      },
+      operationInfo
+    ) => {
+      if (
```

Comment:
this reduce is to complicated. Could you try to restructure the code by introducind a type for the 
```
{
        uniqueOperationNames: string[];
        operationsWithUniqueNames: OperationInfo[];
        operationsWithDuplicateNames: OperationInfo[];
}
```

introducing helper methods because I got lost in the `if` `followed by multiple `||`

Reply:
Improved by a lot ;)